src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php

```diff
@@ -35,19 +35,28 @@ class GuardAuthenticatorHandler
     private $tokenStorage;
     private $dispatcher;
     private $sessionStrategy;
+    private $statelessProviderKeys;
 
-    public function __construct(TokenStorageInterface $tokenStorage, EventDispatcherInterface $eventDispatcher = null)
+    /**
+     * @param TokenStorageInterface         $tokenStorage
+     * @param EventDispatcherInterface|null $eventDispatcher
+     * @param array                         $statelessProviderKeys An array of provider/firewall keys that are "stateless" and so do not need the session migrated on success
+     */
+    public function __construct(TokenStorageInterface $tokenStorage, EventDispatcherInterface $eventDispatcher = null, array $statelessProviderKeys = array())
     {
         $this->tokenStorage = $tokenStorage;
         $this->dispatcher = $eventDispatcher;
+        $this->statelessProviderKeys = $statelessProviderKeys;
     }
 
     /**
      * Authenticates the given token in the system.
      */
     public function authenticateWithToken(TokenInterface $token, Request $request)
     {
-        $this->migrateSession($request, $token);
+        $providerKey = func_num_args() > 2 ? func_get_arg(2) : null;
```

Comment:
\func_num_args()
The new argument should be added on the signature between /**/, and on the docblock.